web3.js/src/connection.ts

```diff
@@ -2120,6 +2120,18 @@ export type ConnectionConfig = {
   confirmTransactionInitialTimeout?: number;
 };
 
+function warnListenerRemovedWithoutActiveSubscription(
+  id: number,
+  label: string,
+) {
+  console.warn(
+    `You removed a(n) ${label} subscription with id \`${id}\` ` +
+      'where no active subscription having that id could be . ' +
+      'found. Verify that your code does not try to ' +
+      'unsubscribe the same listener more than once.',
+  );
```

Comment:
```suggestion
  console.warn(
    `Ignored unsubscribe request because an active subscription with id \`${id}\` for \'${label}\' events could not be found.`);
```
I made some adjustments and here's why:
1. We aren't necessarily just talking to the developer here 😉, it could show up in a user's console
2. We should be clear that we just ignored this request and that nothing bad happened
3. We don't need to hand hold devs here, a warning stating what happened should be enough for them to debug what happened. They could have also passed an incorrect id or called the wrong unsubscribe method.

Feel free to take or leave them, don't want to bike shed this too much :)

Reply:
Love it.